test/amqp-integration-test.py

```diff
@@ -69,57 +48,47 @@ def run_node_test():
         die(ExitStatus.Error)
     if subprocess.Popen('''
         node test.js --email foo@letsencrypt.org --agree true \
-          --domains foo.com --new-reg http://localhost:4300/acme/new-reg \
+          --domains foo.com --new-reg http://localhost:4000/acme/new-reg \
           --certKey %s/key.pem --cert %s/cert.der
         ''' % (tempdir, tempdir), shell=True).wait() != 0:
         print("\nIssuing failed")
         die(ExitStatus.NodeFailure)
     if subprocess.Popen('''
-        node revoke.js %s/cert.der %s/key.pem http://localhost:4300/acme/revoke-cert
+        node revoke.js %s/cert.der %s/key.pem http://localhost:4000/acme/revoke-cert
         ''' % (tempdir, tempdir), shell=True).wait() != 0:
         print("\nRevoking failed")
         die(ExitStatus.NodeFailure)
 
     return 0
 
+
 def run_client_tests():
     root = os.environ.get("LETSENCRYPT_PATH")
     assert root is not None, (
         "Please set LETSENCRYPT_PATH env variable to point at "
         "initialized (virtualenv) client repo root")
-    os.environ['SERVER'] = 'http://localhost:4300/acme/new-reg'
+    os.environ['SERVER'] = 'http://localhost:4000/acme/new-reg'
     test_script_path = os.path.join(root, 'tests', 'boulder-integration.sh')
     if subprocess.Popen(test_script_path, shell=True, cwd=root).wait() != 0:
         die(ExitStatus.PythonFailure)
 
-try:
-    start()
-    busted = []
-    for pinfo in processes:
-        if pinfo.proc.poll() is not None:
-            busted.append(pinfo)
-    if len(busted) != 0:
-        print "\n\nThese processes didn't start up successfully (check above for their output):"
-        for pinfo in busted:
-            print "\t'%s' exited with %d" % (pinfo.cmd, pinfo.proc.returncode)
-        sys.exit(1)
-
-    run_node_test()
-    run_client_tests()
-except Exception as e:
-    exit_status = ExitStatus.Error
-    print e
-finally:
-    for pinfo in processes:
-        if pinfo.proc.poll() is None:
-            pinfo.proc.kill()
-        else:
-            exit_status = 1
 
+@atexit.register
+def cleanup():
+    import shutil
     shutil.rmtree(tempdir)
-
-    if exit_status == 0:
+    if exit_status == ExitStatus.OK:
         print("\n\nSUCCESS")
     else:
         print("\n\nFAILURE")
-    sys.exit(exit_status)
+
+
+exit_status = ExitStatus.OK
+tempdir = tempfile.mkdtemp()
+os.environ['GORACE'] = 'halt_on_error=1'
```

Comment:
Yes, that's exactly why I ended up here, and yes, setting it only in the "go build" process is much nicer. Updated.